### PR TITLE
refactor: common input interface for python and pynative models 

### PR DESCRIPTION
**Pull request by shekhar-rudder**
## Description of the change

Currently, we are passing different values for `inputs` in python vs pynative models. For python, it is a selector sql and for pynative, it is the material name. This was also reflected in the name of the variable `input_material_or_selector_sql`

- In this PR, a common input interface is being created in pythonWHT and pyNativeWHT which is being passed to train and predict modules.
- Replaced `extract_seq_no_from_select_query` util with a method in whtService `get_latest_seq_no`
- Moved logic of `get_input_models` to `get_inputs` function
- Moved `site_config_path` and `project_folder_path` from `init` function of `whtService` to the constructor. This was done since now the mentioned fields are required before calling the `init` method. Renamed `init` to `set_connector`
- Removed `get_latest_entity_var_table_name` from `RudderPB` file since it was unused

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/predictions/profiles_mlcorelib/wht/pythonWHT.py [hunk 1]
```diff
@@ -579,6 +503,64 @@ def get_complete_sequences(sequences: List[Sequence]) -> int:
             feature_data_min_date_diff,
         )
 
+    def get_latest_seq_no(self, inputs: List[dict]) -> int:
+        return int(inputs[0]["table_name"].split("_")[-1])
+
+    def get_inputs(self, selector_sqls: str, skipCompile: bool) -> List[dict]:
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/wht/pythonWHT.py:509]: can you rename skipCompile -> skip_compile

File: src/predictions/profiles_mlcorelib/py_native/prediction.py [hunk 2]
```diff
@@ -116,25 +116,21 @@ def _get_train_output_filepath(self, this: WhtMaterial):
     def execute(self, this: WhtMaterial):
         logger.set_logger(self.logger)
         site_config_path = this.wht_ctx.site_config().get("FilePath")
-        whtService = PyNativeWHT(this)
+        whtService = PyNativeWHT(this, "", "")
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/prediction.py:119]: Cant we pass None, None instead of empty strings?

**Review by dpatchigolla — APPROVED** [review]
minor rename suggestions

File: src/predictions/profiles_mlcorelib/wht/pythonWHT.py (hunk 1 above)
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/wht/pythonWHT.py]: renamed

File: src/predictions/profiles_mlcorelib/py_native/prediction.py (hunk 2 above)
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/prediction.py]: Replaced empty string with None

**Pull request merged**